### PR TITLE
Implement searchable multiselect widget

### PR DESCRIPTION
**Pull request by ellykits**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes https://github.com/opensrp/fhircore/issues/3068

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ellykits** [comment]
Drafted this PR to showcase the WIP for the new location selector UI

**Comment by ellykits** [comment]
This is not the complete UI, the rest of the work is still under active development. 

@dubdabasoduba verify the UX

[Screencast from 03-05-2024 05:31:55 PM.webm](https://github.com/opensrp/fhircore/assets/26865532/3dbbe4f2-5c0f-417f-bfb8-c5da7cf5ac5e)

**Comment by ellykits** [comment]
This implementation should work with data from any Resource. The UI can also be partially customized.

File: android/engine/src/test/java/org/smartregister/fhircore/engine/sync/SyncListenerManagerTest.kt [hunk 1]
```diff
@@ -62,6 +59,8 @@ class SyncListenerManagerTest : RobolectricTest() {
         configService = configService,
         sharedPreferencesHelper = sharedPreferencesHelper,
         configurationRegistry = configurationRegistry,
+        context = ApplicationProvider.getApplicationContext(),
+        dispatcherProvider = co,
```
Comment (hamza-vd) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/sync/SyncListenerManagerTest.kt:63]: @ellykits the `dispatcherProvider` value `co` is an unresolved variable, Could you please guide the correct value here?

**Comment by ellykits** [comment]
More tests to be added later, issue for tracking https://github.com/opensrp/fhircore/issues/3253

**Pull request merged**